### PR TITLE
fix some bugs

### PR DESCRIPTION
**Pull request by iamlinjunhong**
## What type of PR is this?

- [ ] API-change
- [ ] BUG
- [ ] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #6151 #5995 #5618

## What this PR does / why we need it:

### CODE REVIEW

**Review by fengttt — COMMENTED** [review]
Need bvt tests.

Is there a way to test insert result?

**Comment by iamlinjunhong** [comment]
> Need bvt tests.
> 
> Is there a way to test insert result?

There is no way to test insert result in bvt test

**Review by aunjgr — APPROVED** [review]

File: pkg/sql/colexec/insert/insert.go [hunk 1]
```diff
@@ -71,8 +72,8 @@ func handleWrite(n *Argument, proc *process.Process, ctx context.Context, bat *b
 		b.Clean(proc.Mp())
 	}
 	err := n.TargetTable.Write(ctx, bat)
+	atomic.AddUint64(&n.Affected, uint64(bat.Vecs[0].Length()))
```
Comment (nnsgmsone) [inline pkg/sql/colexec/insert/insert.go:75]: Should  increase this value even in case of error?

Comment (iamlinjunhong) [reply pkg/sql/colexec/insert/insert.go:78]: done

**Review by nnsgmsone — APPROVED** [review]

**Pull request merged**